src/test/java/edu/wpi/grip/core/ConnectionTest.java

```diff
@@ -14,7 +14,11 @@
     @Test
     public void testInputSocketChanges() {
         Socket<Double> foo = new Socket<>(eventBus, fooHint);
+        foo.setDirection(Socket.Direction.OUTPUT);
+
         Socket<Double> bar = new Socket<>(eventBus, barHint);
+        bar.setDirection(Socket.Direction.INPUT);
```

Comment:
So explicitly say that it's an input socket.  For the purposes of the unit test, I want to manually create a connection, which must be between an input socket and an output socket.